src/modules/NtpClient.rb

```diff
@@ -648,13 +525,13 @@ def IsRandomServersServiceEnabled
 
       Builtins.foreach(GetUsedNtpServers()) do |used_server|
         # if server is needed by pool.ntp.org and matches
-        if Ops.get(needed_servers, used_server) != nil
+        if !Ops.get(needed_servers, used_server).nil?
           Ops.set(needed_servers, used_server, true)
         end
       end
 
       ret = true
-      Builtins.foreach(needed_servers) do |nserver_name, ns_value|
+      Builtins.foreach(needed_servers) do |_nserver_name, ns_value|
```

Comment:
Some of the changes are just for make rubocop happier. There is a lot of methods that could be rewrite but I’m trying to add tests before touch or clean some method.